custom_components/tahoma/tahoma_device.py

```diff
@@ -67,20 +65,19 @@ def assumed_state(self):
     @property
     def device_state_attributes(self):
         """Return the state attributes of the device."""
-
         attr = {
             "uiclass": self.tahoma_device.uiclass,
             "widget": self.tahoma_device.widget,
             "type": self.tahoma_device.type,
         }
 
-        if CORE_RSSI_LEVEL_STATE in self.tahoma_device.active_states:
-            attr[ATTR_RSSI_LEVEL] = self.tahoma_device.active_states[
-                CORE_RSSI_LEVEL_STATE
-            ]
+        states = self.tahoma_device.active_states
+
+        if CORE_RSSI_LEVEL_STATE in states:
+            attr[ATTR_RSSI_LEVEL] = states.get(CORE_RSSI_LEVEL_STATE)
```

Comment:
Same remark than above.

Reply:
This one is different, it will only set the rssi level attribute if it exists. It doesn't exist for all Somfy TaHoma devices, thus this should be a dynamic assignment.

Reply:
It would be interesting to see what HA do with NONE attribute. But yes, we can keep it like this.